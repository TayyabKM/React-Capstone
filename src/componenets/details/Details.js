import React from 'react';
import PropTypes from 'prop-types';
import '../../assets/details.css';

function Details(props) {
  const {
    population,
    name,
    flags,
    official,
    unMember,
    languages,
    currencies,
    independent,
    region,
    area,
  } = props;
  return (
    <>
      <div className="row" style={{ margin: 'auto' }}>
        <div className="container text-white" style={{ display: 'flex', alignItems: 'center' }}>
          <h1 style={{ marginRight: '20px', padding: '20px' }}>{name}</h1>
          <img src={flags} alt="Country flag" style={{ marginLeft: 'auto', padding: '20px' }} />
        </div>
      </div>
      <table className="table table-striped text-white">
        <tbody>
          <tr>
            <td>Official Name</td>
            <td>{official}</td>
          </tr>
          <tr>
            <td>Region</td>
            <td>{region}</td>
          </tr>
          <tr>
            <td>Population</td>
            <td>{population}</td>
          </tr>
          <tr>
            <td>Language</td>
            <td>{languages}</td>
          </tr>
          <tr>
            <td>United Nations Member</td>
            <td>{unMember}</td>
          </tr>
          <tr>
            <td>Currencies</td>
            <td>{currencies}</td>
          </tr>
          <tr>
            <td>Independent</td>
            <td>{independent || 'N/A'}</td>
          </tr>
          <tr>
            <td>Area</td>
            <td>
              {area}
              <span>  Square Kilometer</span>
            </td>
          </tr>
        </tbody>
      </table>
    </>
  );
}

Details.propTypes = {
  population: PropTypes.number.isRequired,
  area: PropTypes.number.isRequired,
  flags: PropTypes.string.isRequired,
  official: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  unMember: PropTypes.string.isRequired,
  languages: PropTypes.string.isRequired,
  currencies: PropTypes.string.isRequired,
  independent: PropTypes.string.isRequired,
  region: PropTypes.string.isRequired,
};

export default Details;
