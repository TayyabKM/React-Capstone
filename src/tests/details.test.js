import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import Details from '../componenets/details/Details';

it('renders correctly', () => {
  const props = {
    population: 100000,
    area: 2000,
    flags: 'https://example.com/flag.png',
    official: 'Official Name',
    name: 'Country Name',
    unMember: 'Yes',
    languages: 'Language',
    currencies: 'Currency',
    independent: 'Yes',
    region: 'Region',
  };
  const tree = renderer
    .create(
      <Router>
        <Details
          population={props.population}
          area={props.area}
          flags={props.flags}
          official={props.official}
          name={props.name}
          unMember={props.unMember}
          languages={props.languages}
          currencies={props.currencies}
          independent={props.independent}
          region={props.region}
        />
      </Router>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
