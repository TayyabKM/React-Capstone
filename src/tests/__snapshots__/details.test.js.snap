// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div
    className="row"
    style={
      Object {
        "margin": "auto",
      }
    }
  >
    <div
      className="container text-white"
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <h1
        style={
          Object {
            "marginRight": "20px",
            "padding": "20px",
          }
        }
      >
        Country Name
      </h1>
      <img
        alt="Country flag"
        src="https://example.com/flag.png"
        style={
          Object {
            "marginLeft": "auto",
            "padding": "20px",
          }
        }
      />
    </div>
  </div>,
  <table
    className="table table-striped text-white"
  >
    <tbody>
      <tr>
        <td>
          Official Name
        </td>
        <td>
          Official Name
        </td>
      </tr>
      <tr>
        <td>
          Region
        </td>
        <td>
          Region
        </td>
      </tr>
      <tr>
        <td>
          Population
        </td>
        <td>
          100000
        </td>
      </tr>
      <tr>
        <td>
          Language
        </td>
        <td>
          Language
        </td>
      </tr>
      <tr>
        <td>
          United Nations Member
        </td>
        <td>
          Yes
        </td>
      </tr>
      <tr>
        <td>
          Currencies
        </td>
        <td>
          Currency
        </td>
      </tr>
      <tr>
        <td>
          Independent
        </td>
        <td>
          Yes
        </td>
      </tr>
      <tr>
        <td>
          Area
        </td>
        <td>
          2000
          <span>
              Square Kilometer
          </span>
        </td>
      </tr>
    </tbody>
  </table>,
]
`;
